pkg/initializers/dataset/cache_initializer_test.py

```diff
@@ -0,0 +1,365 @@
+                "train_job_name": "custom-job",
+                "cache_image": "custom-image:latest",
+                "cluster_size": "5",
+                "metadata_loc": "s3://bucket/metadata",
+                "table_name": "test_table",
+                "schema_name": "test_schema",
+                "iam_role": "arn:aws:iam::123456789012:role/custom-role",
+                "head_cpu": "4",
+                "head_mem": "8Gi",
+                "worker_cpu": "8",
+                "worker_mem": "16Gi",
+            },
+            {
+                "storage_uri": "cache://test-uri",
+                "train_job_name": "custom-job",
+                "cache_image": "custom-image:latest",
+                "cluster_size": "5",
+                "metadata_loc": "s3://bucket/metadata",
+                "table_name": "test_table",
+                "schema_name": "test_schema",
+                "iam_role": "arn:aws:iam::123456789012:role/custom-role",
+                "head_cpu": "4",
+                "head_mem": "8Gi",
+                "worker_cpu": "8",
+                "worker_mem": "16Gi",
+            },
+        ),
+        (
+            "Minimal config with only storage_uri",
+            {"storage_uri": "cache://minimal-uri"},
+            {
+                "storage_uri": "cache://minimal-uri",
+                "train_job_name": None,
+                "cache_image": None,
+                "cluster_size": None,
+                "metadata_loc": None,
+                "table_name": None,
+                "schema_name": None,
+                "iam_role": None,
+                "head_cpu": None,
+                "head_mem": None,
+                "worker_cpu": None,
+                "worker_mem": None,
+            },
+        ),
+        (
+            "Partial config with some values",
+            {
+                "storage_uri": "cache://partial-uri",
+                "train_job_name": "partial-job",
+                "head_cpu": "2",
+                "worker_cpu": "4",
+            },
+            {
+                "storage_uri": "cache://partial-uri",
+                "train_job_name": "partial-job",
+                "cache_image": None,
+                "cluster_size": None,
+                "metadata_loc": None,
+                "table_name": None,
+                "schema_name": None,
+                "iam_role": None,
+                "head_cpu": "2",
+                "head_mem": None,
+                "worker_cpu": "4",
+                "worker_mem": None,
+            },
+        ),
+    ],
+)
+def test_load_config(test_name, test_config, expected):
+    """Test config loading with different configurations"""
+    print(f"Running test: {test_name}")
+
+    cache_initializer_instance = CacheInitializer()
+
+    with patch.object(utils, "get_config_from_env", return_value=test_config):
+        cache_initializer_instance.load_config()
+        assert cache_initializer_instance.config.storage_uri == expected["storage_uri"]
```

Comment:
I think, we can simplify this by using `cache_initializer_instance.config.__dict__`, since it represents the `expected` dict.

Reply:
@akshaychitneni Did you check this ?